src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs

```diff
@@ -93,8 +93,10 @@ static void CreateIndexes(MongoPersistenceProvider instance)
             if (!indexesCreated)
             {
                 instance.WorkflowInstances.Indexes.CreateOne(new CreateIndexModel<WorkflowInstance>(
-                    Builders<WorkflowInstance>.IndexKeys.Ascending(x => x.NextExecution),
-                    new CreateIndexOptions {Background = true, Name = "idx_nextExec"}));
+                    Builders<WorkflowInstance>.IndexKeys
```

Comment:
In our experiment, we found that the following index works better 

```
 instance.WorkflowInstances.Indexes.CreateOne(new CreateIndexModel<WorkflowInstance>(
                    Builders<WorkflowInstance>.IndexKeys
                        .Ascending(x => x.NextExecution)
                        .Ascending(x => x.Status)
                        .Ascending(x => x.Id),
                    new CreateIndexOptions {Background = true, Name = "idx_nextExec_Status_Id)"}));
```

Reply:
Nice catch, thanks!
Just did a test and both run on index but I missed the projection.
![image](https://user-images.githubusercontent.com/13761704/204080898-4193a069-e1c3-49a1-ad49-d537942ec76a.png)